app/models/ott/CategorisationInfo.scala

```diff
@@ -88,22 +89,24 @@ object CategorisationInfo {
         val category2Assessments = assessmentsSorted.filter(ass => ass.isCategory2)
 
         val category1ToAnswer = category1Assessments.filter(ass => !ass.hasNoAnswers).filter(ass => !ass.isNiphlsAnswer)
-        val category2ToAnswer = category2Assessments.filter(ass => !ass.hasNoAnswers)
+        val category2ToAnswer = category2Assessments.filter(ass => !ass.hasNoAnswers).filter(ass => !ass.isNirmsAnswer)
 
         val areAllCategory1Answerable = category1ToAnswer.size == category1Assessments.size
         val areAllCategory2Answerable = category2ToAnswer.size == category2Assessments.size
 
         val isNiphlsAssessment =
           category1Assessments.exists(ass => ass.isNiphlsAnswer) && category2Assessments.exists(ass => ass.hasNoAnswers)
 
+        val isNirmsAssessment = category2Assessments.exists(ass => ass.isNirmsAnswer)
+
         val questionsToAnswer = {
           if (isNiphlsAssessment && traderProfile.niphlNumber.isDefined) {
             category1ToAnswer
           } else if (isNiphlsAssessment) {
             Seq.empty
           } else if (!areAllCategory1Answerable) {
             Seq.empty
-          } else if (!areAllCategory2Answerable) {
+          } else if (!isNirmsAssessment && !areAllCategory2Answerable) {
```

Comment:
Additional case to consider:

Category 1 questions + Category 2 questions + Nirms question

Currently if trader does not have Nirms, it makes them answer all the Category 2 questions, but this is a waste of time because the answer cannot be Category 3.

Reply:
Logic updated